packages/typespec-client-generator-core/generated-defs/Azure.ClientGenerator.Core.Legacy.ts

```diff
@@ -0,0 +1,46 @@
+import type { DecoratorContext, Model } from "@typespec/compiler";
+
+/**
+ * Adds support for client-level multiple levels of inheritance.
+ *
+ * This decorator will update the models returned from TCGC to include the multi-level inheritance information.
+ *
+ * This decorator is considered legacy functionality and may be deprecated in future releases.
+ *
+ * @param target The target model that will gain legacy inheritance behavior
+ * @param value The model whose properties should be inherited from
+ * @param scope Optional parameter to specify which language emitters this applies to
+ * @example Three-level inheritance
+ *
+ * ```typespec
+ * @discriminator("kind")
```

Comment:
Can we make this example a little bit more descriptive? For example, it would be super helpful if it illustrates clearly what is the user intent and how the new decorator enables that. From looking at the TypeSpec here I don't think that is clear.